src/clj/puppetlabs/services/config/puppet_server_config_service.clj

```diff
@@ -26,7 +26,9 @@
           "Initializing with the following settings from core Puppet:\n%s"
           (ks/pprint-to-string puppet-config))
         (core/init-webserver! override-webserver-settings!
-                              (get-in tk-config [:webserver])
+                              (get-in tk-config
+                                      [:webserver :puppet-server]
+                                      (get-in tk-config [:webserver]))
                               puppet-config)
```

Comment:
There's also the issue of the different configuration styles allowed for webserver configuration, as well as the presence of a default webserver, meaning the user does not always need to explicitly state which webserver they want a route added to. Parsing out the configuration for the webserver that Puppet Server is adding routes to is doable, but it requires a lot of specific knowledge about how the webserver configuration is structured, which makes me think that if we were to do that it should land in tk-webserver-jetty9 as a new service function rather than here in Puppet server. 

Saying that, we'd still have the circular dependency issue, which I don't have a solution for.